projects/index.mjs

```diff
@@ -0,0 +1,105 @@
+/** Route handler for the /projects endpoint */
+import express from 'express'
+import * as logic from './projects.mjs'
+import * as utils from '../utilities/shared.mjs'
+import cors from 'cors'
+import auth0Middleware from '../auth/index.mjs'
+import 'dotenv/config'
+
+let router = express.Router()
+router.use(
+  cors({
+    methods: 'GET',
+    allowedHeaders: [
+      'Content-Type',
+      'Content-Length',
+      'Allow',
+      'Authorization',
+      'Location',
+      'ETag',
+      'Connection',
+      'Keep-Alive',
+      'Date',
+      'Cache-Control',
+      'Last-Modified',
+      'Link',
+      'X-HTTP-Method-Override',
+    ],
+    exposedHeaders: '*',
+    origin: '*',
+    maxAge: '600',
+  })
+)
+
+/**
+ * Validates queries for /projects endpoint.
+ * 
+ * @param queries A JSON object of query parameters for the project list lookup.
+ * @return A boolean indicating whether validations passed or failed.
+ */
+function validateQueries(queries) {
+
+}
+
+/**
+ * Authenticated endpoint to return a list of a user's projects.
+ * 
+ * @param options A JSON object of options for the project list lookup.
+ * @return An unexpanded list of the user's projects. Returns empty list if none found.
+ */
+export async function respondWithProjects(user, options, res){
+  // Set option defaults
+  let hasRoles       = options.hasRoles       ?? 'ALL' 
```

Comment:
If I'm reading this right, these `let`s are not needed and should be `const`. In fact, the one's that change (`NOW` to `Date.now()`) should be preset and then the defaults can be left alone.

Reply:
The idea I was going for is that the only valid inputs for this parameter are either "NOW" or a date, and any other input would return an error. I thought this would make it easier to validate the queries, since I would only have to do one check for "NOW" in the same place I check for a valid date.

Reply:
My thought process, at least for the createdBefore and modifiedBefore parameters, is that the valid inputs would be either "NOW" or a numerical date. Even if I make Date.now() the default here, I'd have to ensure that if the user manually sends "NOW" then that would be accepted. That would require either changing the createdBefore variable to Date.now(), or creating an entirely new variable to use when actually filtering through the projects[] array. Using `let` seemed cleaner. I'll change the others, though, since those shouldn't change.